### ISSUE TITLE
NVDA with braille display, freeze with Chrome

### ISSUE THREAD

**Issue by rperez030** [nvaccess/nvda#11247]
### Steps to reproduce:
Browse the web normally using Google Chrome. The problem seems more likely to manifest if a braille display is connected.
### Actual behavior:
NVDA gets frozen. Killing the process seems to be the only way out. If the speech viewer is opened the last text spoken by NVDA will appear many times.
### Expected behavior:
NVDA should behave normally.
#### NVDA installed/portable/running from source:
Portable.
#### NVDA version:
NVDA 2020.1
#### Windows version:
Version	10.0.18363 Build 18363

#### Name and version of other software in use when reproducing the issue:
Google chrome Version 83.0.4103.97 (Official Build) 
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
NVDA 2019.3.x
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

I have observed this problems in two computers always with a braille display connected. It is hard to tell whether the cause is always the same because the only common  factor seems to be a braille display. I am attaching a log file. An unhandled exception can be found when the problem occurred.
[nvda failure.log](https://github.com/nvaccess/nvda/files/4760960/nvda.failure.log)

**Comment by feerrenrut** [comment]
Braille driver: `Loaded braille display driver brailliantB, current display has 40 cells`
Relevant error in log:
```
ERROR - eventHandler.executeEvent (08:12:47.082) - MainThread (5568):
error executing event: gainFocus on <NVDAObjects.Dynamic_IAccessibleRichEdit20WindowNVDAObject object at 0x072270F0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\behaviors.pyc", line 185, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 1103, in event_gainFocus
  File "braille.pyc", line 1905, in handleGainFocus
  File "braille.pyc", line 1926, in _doNewObject
  File "braille.pyc", line 1969, in scrollToCursorOrSelection
  File "braille.pyc", line 1370, in scrollTo
  File "braille.pyc", line 1250, in regionPosToBufferPos
LookupError: No such position
ERROR - core.CorePump.run (08:12:47.280) - MainThread (5568):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyc", line 517, in run
  File "braille.pyc", line 2216, in pumpAll
  File "braille.pyc", line 1952, in handlePendingCaretUpdate
  File "braille.pyc", line 1961, in _doCursorMove
  File "braille.pyc", line 1969, in scrollToCursorOrSelection
  File "braille.pyc", line 1370, in scrollTo
  File "braille.pyc", line 1250, in regionPosToBufferPos
LookupError: No such position
```

@leonardder are you interested in looking into this?

**Comment by rperez030** [comment]
I want to add that I am also observing the problem in Outlook when composing an email. It manifest in exactly the same way. I am moving the cursor and NVDA freezes. Even Narrator becomes sluggish  and Outlook becomes unresponsive until NVDA is killed. It does not seem to occur if I am not using a braille display.

**Comment by bhavyashah** [comment]
@leonardder  This is a friendly reminder to consider responding to https://github.com/nvaccess/nvda/issues/11247#issuecomment-642616633 and share if you might be able to investigate this issue.
Also, this is a gentle request to developers to briefly document the scope of this ticket. Does the reported freeze seem specific to Braille and/or Chrome or is it wider? Understandably, the details will be known only after a thorough diagnosis, but in case there are any preliminary conditions that are known, please share. I think this may help @rperez030  and others better judge whether to report their similar freezes as a comment here or via a separate ticket.

**Comment by michaelDCurran** [comment]
@rperez030 Do you have any add-ons in use? If so, does the issue go away with all add-ons disabled?

Another concerning part of the log is:
```
ERROR - unhandled exception (09:58:41.374) - winInputHook (19524):
Traceback (most recent call last):
  File "wx\core.pyc", line 3240, in <lambda>
wx._core.wxAssertionError: C++ assertion "wxThread::IsMain()" failed at ..\..\src\common\timerimpl.cpp(60) in wxTimerImpl::Start(): timer can only be started from the main thread
```

**Comment by michaelDCurran** [comment]
@rperez030 Are you still able to reproduce this with NVDA 2020.3 betas or latest alphas?

**Comment by Adriani90** [comment]
We didn't get any updates from the initial author since many years, I am closing this as abandoned. Please comment on it if you are still having this issue and we can reopen.
